lib/plugins/plugin.rb

```diff
@@ -19,7 +20,7 @@ class Plugin
   attr_reader :help, :type
   attr_accessor :config
 
-  def initialize(file, settings)
+  def initialize(file, settings, port)
```

Comment:
There is a lot of indirection going on here to set port->@dynamic_port->then later to set @port. Is there possibly a more straight forward way to do this?